### DISCUSSION TITLE
YoloV8 Training Slower than YoloV5?

### DISCUSSION THREAD

**Discussion by ajrenzo** [ultralytics/ultralytics#1819] [Q&A] [answered]
I'm trying to train a custom dataset on yolov8n.pt. I used to use yolov5 until I saw V8 was out and decided to try it. My first attempt at training the dataset took over 1200 minutes, while training on yolov5 only took around 200. The yolov5 format looks as such:
`!cd yolov5 && python train.py --cache --img 200 --batch 500 --epochs 2000 --data dataset_2.yml --weights yolov5n.pt`

The V8 training code is here:
```
from ultralytics import YOLO

model = YOLO('yolov8n.pt')
model.train(data='F:\Documents\Training\Pytorch\yolov5\dataset_v8.yaml' , batch=500, cache=True, epochs=1000, imgsz=200)
```
Even on 1000 epochs V8 takes longer than 2000 epochs of V5. I noticed that the V8 takes some time to start up each epoch (GPU spikes take longer between each set), which it then trains quickly. The V5 is much quicker to start up the epochs for training. Is there a setting I'm forgetting for V8 to have the epochs start faster?

**Comment by ajrenzo** [comment]
Found a temporary solution another thread relating to the long epoch time: Set workers=0. From what I can tell its a windows bug. Might also use linux to make things faster until a solution for this has been found.

**Reply by glenn-jocher** [reply, marked as answer]
@ajrenzo this should be resolved in the latest version of YOLOv8, which utilizes the faster InfiniteDataloader by default now. You can update your package with `pip install -U ultralytics`
